WordPress/src/main/java/org/wordpress/android/ui/reader/actions/ReaderBlogActions.java

```diff
@@ -80,9 +80,17 @@ public void onResponse(JSONObject jsonObject) {
             public void onErrorResponse(VolleyError volleyError) {
                 AppLog.w(T.READER, "blog " + actionName + " failed with error");
                 AppLog.e(T.READER, volleyError);
-                localRevertFollowBlogId(blogId, isAskingToFollow);
-                if (actionListener != null) {
-                    actionListener.onActionResult(false);
+                // check if we get a 403 when unfollowing - this will happen when we attempt
+                // to unfollow a blog that no longer exists - the workaround is to unfollow
+                // by url - note that the v1.2 endpoint will return a 404 in this situation
+                int status = VolleyUtils.statusCodeFromVolleyError(volleyError);
+                if (status == 403 && !isAskingToFollow) {
+                    internalUnfollowBlogByUrl(blogId, actionListener);
```

Comment:
I think the fact that the action listener is passed to `internalUnfollowBlogByUrl` makes it clear that it will take care of calling it. I'd rather not add another comment (seems like more noise) but it's not a big deal to me and I'll do it if you still feel it's necessary.